Libraries/LibC/stdio.cpp

```diff
@@ -911,25 +910,43 @@ void perror(const char* s)
 static int parse_mode(const char* mode)
 {
     int flags = 0;
+
     // NOTE: rt is a non-standard mode which opens a file for read, explicitly
     // specifying that it's a text file
-    if (!strcmp(mode, "r") || !strcmp(mode, "rb") || !strcmp(mode, "rt"))
-        flags = O_RDONLY;
-    else if (!strcmp(mode, "r+") || !strcmp(mode, "rb+"))
-        flags = O_RDWR;
-    else if (!strcmp(mode, "w") || !strcmp(mode, "wb"))
-        flags = O_WRONLY | O_CREAT | O_TRUNC;
-    else if (!strcmp(mode, "w+") || !strcmp(mode, "wb+"))
-        flags = O_RDWR | O_CREAT | O_TRUNC;
-    else if (!strcmp(mode, "a") || !strcmp(mode, "ab"))
-        flags = O_WRONLY | O_APPEND | O_CREAT;
-    else if (!strcmp(mode, "a+") || !strcmp(mode, "ab+"))
-        flags = O_RDWR | O_APPEND | O_CREAT;
-    else {
-        dbg() << "Unexpected mode _" << mode << "_";
-        ASSERT_NOT_REACHED();
+    auto length = strlen(mode);
+    for (size_t i = 0; i < length; ++i) {
+        switch (mode[i]) {
+        case '\0':
+            return flags;
```

Comment:
(I think this is the last nitpick!)

Since we return when we see the `'\0'`, we don't have to pre-calculate length. I'd just do

```c++
for (; *mode; mode++)
    ....
return flags;
```
But feel free to make it a `while` loop if you would prefer that

Reply:
This one I can happily use `for` for (for for for for...)
It's just the multiline `for`'s that make me want to switch to a `while` loop

Reply:
> (I think this is the last nitpick!)

It's not, the comment at the end is wrong now lol

Reply:
I was hoping you'd catch that one yourself :wink:

Reply:
(and you don't need the `case '\0'` here anymore)

Reply:
I'll go ahead and add it back so you can add one more nitpick teehee

Reply:
> and you don't need the `case '\0'` here anymore

Bump

Reply:
oh. 
sorry, didn't notice that.